### ISSUE TITLE
`manual_is_multiple_of` provides a broken suggestion for negated modulo checks

### ISSUE THREAD

**Issue by cachebag** [rust-lang/rust-clippy#17419]
### Summary

The `clippy::manual_is_multiple_of` lint incorrectly suggests a fix for expressions like `x % y != 0`. Instead of replacing the entire expression, it prepends a `!` to the variable, resulting in broken code (`!x % y != 0`). This broken code then triggers the same lint again with an even more confusing suggestion IMHO, due to operator precedence issues.

### Lint Name

clippy::manual_is_multiple_of

### Reproducer

I tried this code:

```rust
fn main() {
    let some_len = 10;
    let record_size = 3;

    // Original code that triggers the lint
    if some_len % record_size != 0 {
        println!("Not a multiple");
    }
}
```

I saw this happen:

First, Clippy gives the following warning:
```
error: manual implementation of `.is_multiple_of()`
   --> src/main.rs:6:8
    |
6   |     if some_len % record_size != 0 {
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with: `!some_len.is_multiple_of(record_size)`
```

Applying this suggestion results in `if !some_len % record_size != 0`, which then produces a second, more confusing error:

```
error: manual implementation of `.is_multiple_of()`
   --> src/main.rs:6:8
    |
6   |     if !some_len % record_size != 0 {
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with: `!(!some_len).is_multiple_of(record_size)`
```

I expected to see this happen:

The lint's automatic suggestion should replace the entire expression `some_len % record_size != 0` with `!some_len.is_multiple_of(record_size)`.

The current behavior, which leads to a syntax error and a confusing follow-up lint, is incorrect.

### Version

```text
rustc 1.96.0 (ac68faa20 2026-05-25)
binary: rustc
commit-hash: ac68faa20c58cbccd01ee7208bf3b6e93a7d7f96
commit-date: 2026-05-25
host: x86_64-unknown-linux-gnu
release: 1.96.0
LLVM version: 22.1.2
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by cachebag** [comment]
To be clear, my actual code does something like `sample.data.len()`, which returns a usize (which is unsigned). The lint may not necessarily trigger in my example reproducer.

**Comment by Jarcho** [comment]
The lint does suggests using `!some_len.is_multiple_of(record_size)` as you can see in the diagnostic. I don't know what you're using to apply the suggestion, but `cargo clippy --fix` works correctly.

**Comment by cachebag** [comment]
> I don't know what you're using to apply the suggestion, but cargo clippy --fix works correctly.

Whoops...yea that is on me. Sorry!
